WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionWebViewClient.java

```diff
@@ -0,0 +1,137 @@
+package org.wordpress.android.ui;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import org.wordpress.android.analytics.AnalyticsTracker;
+import org.wordpress.android.fluxc.model.SiteModel;
+import org.wordpress.android.fluxc.store.AccountStore;
+import org.wordpress.android.fluxc.store.SiteStore;
+import org.wordpress.android.login.LoginMode;
+import org.wordpress.android.ui.accounts.LoginActivity;
+import org.wordpress.android.ui.prefs.AppPrefs;
+import org.wordpress.android.util.AppLog;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+
+class JetpackConnectionWebViewClient extends WebViewClient {
+
+    private static final String LOGIN_PATH = "/wp-login.php";
+    private static final String ADMIN_PATH = "/wp-admin/admin.php";
+    private static final String REDIRECT_PARAMETER = "redirect_to=";
+    private static final int REQUEST_CODE = 1;
+    private static final String WORDPRESS_COM_HOST = "wordpress.com";
+    private static final String LOG_IN_PATH = "/log-in";
+    private static final String JETPACK_PATH = "/jetpack";
+    private static final String WORDPRESS_COM_PREFIX = "https://wordpress.com";
+    private static final Uri JETPACK_DEEPLINK_URI = Uri.parse(WPWebViewActivity.JETPACK_CONNECTION_DEEPLINK);
+    private static final String REDIRECT_PAGE_STATE_ITEM = "redirectPage";
+    private static final String FLOW_FINISHED = "FLOW_FINISHED";
+
+    private Activity activity;
+    private final AccountStore accountStore;
+    private final SiteStore mSiteStore;
+
+    private String redirectPage;
+    private boolean flowFinished = false;
+
+    JetpackConnectionWebViewClient(Activity activity, AccountStore accountStore, SiteStore mSiteStore) {
+        this.activity = activity;
```

Comment:
Rather than pass the stores as parameters, we can just use:

```
@Inject AccountStore mAccountStore;
@Inject SiteStore mSiteStore;
```

Reply:
The JetpackConnectionWebViewClient is not an injected field i the activity. I can create an injected factory in order to inject the params. Is that a pattern we use?